docs/install/knative-with-kourier.md

```diff
@@ -0,0 +1,162 @@
+---
+title: "Installing Knative with Kourier"
+linkTitle: "Kourier Ingress and Knative"
+weight: 15
+type: "docs"
+---
+
+[Kourier](https://github.com/3scale/kourier) is an open-source lightweight Knative Ingress based on Envoy. It's been designed for Knative, without requiring any additional custom resource definitions (CRDs).
+
+This guide walks you through the installation of the latest version of Knative
+with Kourier as the ingress.
+
+## Before you Begin
+
+Knative requires a Kubernetes cluster v1.14 or newer, as well as a compatible `kubectl`. This guide assumes that you have already [created a Kubernetes cluster](https://kubernetes.io/docs/setup/) and are using
+bash in a Mac or Linux environment.
+
+## Install Knative
+
+Let's do a core install of Knative Serving with the released yaml templates:
+
+1.  To install Knative, first install the CRDs by running the following `kubectl apply`
+    command. This prevents race conditions during the install, which cause intermittent errors:
+
+        kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-crds.yaml
+
+1.  To complete the install of Knative and its dependencies, next run the
+    following `kubectl apply` command:
+
+        kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-core.yaml
+
+1.  Monitor the Knative Serving namespace and wait until all of the pods come up with a
+    `STATUS` of `Running`:
+
+    ```
+    kubectl get pods -w -n knative-serving
+    ```
+
+
+## Install Kourier
+
+Knative default installation uses Istio to handle internal and external traffic. If you are just interested in exposing Knative applications to the external network, a service mesh adds overhead and increases the system complexity. Kourier provides a way to expose your Knative application in a more simple and lightweight way.
+
+You can install Kourier with `kubectl`:
+
+```
+kubectl apply \
+  --filename https://raw.githubusercontent.com/3scale/kourier/master/deploy/kourier-knative.yaml
```

Comment:
Should this be: https://raw.githubusercontent.com/knative/serving/master/third_party/kourier-latest/kourier.yaml ??

Reply:
diff between the two pages.
```
25c25
<     - name: http2
---
>     - name: http
33,47d32
< apiVersion: v1
< kind: Service
< metadata:
<   name: kourier-tls
<   namespace: kourier-system
< spec:
<   ports:
<     - name: http2
<       port: 443
<       protocol: TCP
<       targetPort: 8443
<   selector:
<     app: 3scale-kourier-gateway
<   type: LoadBalancer
< ---
87,93c72,73
<             httpGet:
<               httpHeaders:
<                 - name: Host
<                   value: internalkourier
<               path: /__internalkouriersnapshot
<               port: 8081
<               scheme: HTTP
---
>             exec:
>               command: ['ash','-c','(printf "GET /__internalkouriersnapshot HTTP/1.1\r\nHost: internalkourier\r\n\r\n"; sleep 1) | nc -n localhost 8081 | grep "HTTP/1.1 200 OK"']
129c109
<         - image: quay.io/3scale/kourier:v0.3.5
---
>         - image: quay.io/3scale/kourier:v0.3.4
202c182,197
<   type: ClusterIP
---
>   type: LoadBalancer
> ---
> apiVersion: v1
> kind: Service
> metadata:
>   name: kourier-external
>   namespace: kourier-system
> spec:
>   ports:
>     - name: http2
>       port: 80
>       protocol: TCP
>       targetPort: 8080
>   selector:
>     app: 3scale-kourier-gateway
>   type: LoadBalancer
216c211
<   type: ClusterIP
---
>   type: LoadBalancer
224a220,225
>     admin:
>       access_log_path: /tmp/test
>       address:
>         socket_address:
>           address: 0.0.0.0
>           port_value: 19000
239,267d239
<       listeners:
<         - name: stats_listener
<           address:
<             socket_address:
<               address: 0.0.0.0
<               port_value: 9000
<           filter_chains:
<             - filters:
<                 - name: envoy.http_connection_manager
<                   config:
<                     stat_prefix: stats_server
<                     route_config:
<                       virtual_hosts:
<                         - name: admin_interface
<                           domains:
<                             - "*"
<                           routes:
<                             - match:
<                                 safe_regex:
<                                   google_re2: {}
<                                   regex: '/(certs|stats(/prometheus)?|server_info|clusters|listeners|ready)?'
<                                 headers:
<                                   - name: ':method'
<                                     exact_match: GET
<                               route:
<                                 cluster: service_stats
<                     http_filters:
<                       - name: envoy.router
<                         config: {}
269,271c241
<         - name: service_stats
<           connect_timeout: 0.250s
<           type: static
---
>         - connect_timeout: 0.2s
273c243
<             cluster_name: service_stats
---
>             cluster_name: xds_cluster
275,285c245,250
<               lb_endpoints:
<                 endpoint:
<                   address:
<                     pipe:
<                       path: /tmp/envoy.admin
<         - name: xds_cluster
<           connect_timeout: 1s
<           hosts:
<             - socket_address:
<                 address: "kourier-control"
<                 port_value: 18000
---
>               - lb_endpoints:
>                 - endpoint:
>                     address:
>                       socket_address:
>                         address: kourier-control
>                         port_value: 18000
286a252,255
>           upstream_connection_options:
>             tcp_keepalive: {}
>           lb_policy: ROUND_ROBIN
>           name: xds_cluster
288,292d256
<     admin:
<       access_log_path: "/dev/stdout"
<       address:
<         pipe:
<           path: /tmp/envoy.admin
```

Reply:
Ah - I see, your version is later.  @samodell may wish to inquire of davidor@ who create the one in knative/serving.